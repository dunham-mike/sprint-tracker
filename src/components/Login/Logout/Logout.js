import React, { Component } from 'react';
import { connect } from 'react-redux';

import * as actions from '../../../store/actions/index';

export class Logout extends Component {
    componentDidMount = () => {
        this.props.onLogout(); // Immediately dispatches the Logout action
    }

    render() {
        return (<div>Logging out...</div>);
    };
}

const mapDispatchToProps = dispatch => {
    return {
        onLogout: () => dispatch(actions.logout())
    }
}

export default connect(null, mapDispatchToProps)(Logout);