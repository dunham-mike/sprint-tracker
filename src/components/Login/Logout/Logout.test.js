import React from 'react';
import { Redirect, BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Enzyme, { mount, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import configureStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';

import { Logout } from './Logout';

Enzyme.configure({ adapter: new Adapter() })

const mountSetup = () => {
    const onLogout = jest.fn();

    const props = {
        onLogout: onLogout
    }

    // keyLength={0} to solve an issue with random keys: https://github.com/ReactTraining/react-router/issues/5579#issuecomment-365780767
    const enzymeMountWrapper = mount(
        <BrowserRouter keyLength={0}>
            <Logout {...props} />
        </BrowserRouter>
    );

    return {
        enzymeMountWrapper,
        onLogout
    }
}

describe('LogOut Component', () => {

    it('should match the snapshot', () => {
        const { enzymeMountWrapper } = mountSetup();
        expect(toJson(enzymeMountWrapper)).toMatchSnapshot();
    });

    it('should render a <Redirect> component', () => {
        const { enzymeMountWrapper } = mountSetup();
        expect(enzymeMountWrapper.contains(<Redirect to="/"/>)).toEqual(true);
    });

    it('should fire onLogout()', () => {
        const { enzymeMountWrapper, onLogout } = mountSetup();
        expect(onLogout).toHaveBeenCalledTimes(1);
    });
});