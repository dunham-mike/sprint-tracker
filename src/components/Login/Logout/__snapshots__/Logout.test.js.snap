// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogOut Component should match the snapshot 1`] = `
<BrowserRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Logout
      onLogout={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
    >
      <div>
        Logging out...
      </div>
    </Logout>
  </Router>
</BrowserRouter>
`;
